pkg/storage/replica_placeholder.go

```diff
@@ -19,8 +19,52 @@ import (
 	"github.com/google/btree"
 )
 
-// ReplicaPlaceholder is created by a Store in anticipation of replacing it at
-// some point in the future with a Replica. It has a RangeDescriptor.
+// ReplicaPlaceholder represents a "lock" of a part of the keyspace on a given
+// *Store for the application of a (preemptive or Raft) snapshot. Placeholders
+// are kept synchronously in two places in (*Store).mu, namely the
+// replicaPlaceholders and replicaByKey maps.
+//
+// Placeholders are inserted in `(*Store).processRaftSnapshotRequest)` when a
```

Comment:
Jeez, this sentence is quite the sandwich. I'm going to leave as is and address it with the review cycle